app/src/main/java/org/oppia/android/app/settings/profile/SnackbarController.kt

```diff
@@ -0,0 +1,39 @@
+package org.oppia.android.app.settings.profile
+
+import androidx.annotation.StringRes
+import org.oppia.android.util.data.DataProvider
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+abstract class SnackbarController @Inject constructor() {
+  // The class should have a queue of snackbars.
```

Comment:
I suggest you use a queue since it is better tailored to getting the earliest item in a more convenient way. Queues are First In First Out which is the kind of behavior we want here. For easy access, I suggest we use a global queue for the class so that it can be easily accessible from all the functions that will need it. 

    // How to declare a queue in this case
    private val snackbarQueue: Queue<SnackbarRequest.ShowSnackbar> = LinkedList()

Reply:
okay